### PR TITLE
Prevent drop zones from blocking cursor events when a drag is not occurring

### PR DESCRIPTION
**Pull request by jonathonherbert**
## What's changed?

Prevent drop zones from blocking cursor events when a drag is not occurring. Drop zones obscure the bottom part of articles, which affects hover and drag operations.

This change gives us a little more freedom to style our drop zones without worrying that they get in the way of other cursor operations, so dropzones for supporting articles have been expanded to be less fiddly, and the space between articles has been reduced by 5px, giving us about an extra article's worth of space on the 900 or so px you get on a MacBook screen 👍 

## Implementation notes

It'd be great to have @RichieAHB's input here to check that my changes are consistent with his lib code!

### General
- [ ] 🤖 Relevant tests added
- [x] ✅ CI checks / tests run locally
- [x] 🔍 Checked on CODE

### Client
- [x] 🚫 No obvious console errors on the client (i.e. React dev mode errors)
- [x] 🎛️ No regressions with existing user interactions (i.e. all existing buttons, inputs etc. work)
- [ ] 📷 Screenshots / GIFs of relevant UI changes included

### CODE REVIEW

File: client-v2/src/lib/dnd/DropZone.tsx [hunk 1]
```diff
@@ -17,10 +18,15 @@ type Props = OuterProps & ContextProps;
 
 interface State {
   isTarget: boolean;
+  isActive: boolean;
 }
 
+const DropZoneContainer = styled.div<{ isActive: boolean }>`
+  ${({ isActive }) => !isActive && 'pointer-events: none'}
```
Comment (jonathonherbert) [inline client-v2/src/lib/dnd/DropZone.tsx]: We could also choose not to render these elements, but this approach minimizes the impact on the DOM. Interested to hear any thoughts.

File: client-v2/src/lib/dnd/Level.tsx [hunk 2]
```diff
@@ -146,6 +146,13 @@ class Level<T> extends React.Component<Props<T>> {
     }
 
     e.preventDefault();
+
+    // Reset the dragging state on drop
```
Comment (RichieAHB) [inline client-v2/src/lib/dnd/Level.tsx]: I think this resetting is handled by the onDragEnd in the Root so needn’t happen here?

File: client-v2/src/lib/dnd/DropZone.tsx [hunk 3]
```diff
@@ -37,10 +43,19 @@ class DropZone extends React.Component<Props, State> {
...
-    return this.props.children(this.state.isTarget);
+    return (
+      <DropZoneContainer isActive={this.state.isActive}>
+        {this.props.children(this.state.isTarget)}
+      </DropZoneContainer>
+    );
   }
 
   private handleStoreUpdate: Sub = (id, hoverIndex) => {
+    const state = this.props.store && this.props.store.getState();
```
Comment (RichieAHB) [inline client-v2/src/lib/dnd/DropZone.tsx]: This is definitely the right way to do this now put potentially some form of PureComponent wrapper might be a bit nicer here in future given we’re just checking for changes everywhere.

File: client-v2/src/lib/dnd/DropZone.tsx (hunk 1 above)
Comment (RichieAHB) [reply client-v2/src/lib/dnd/DropZone.tsx]: This sounds reasonable to me!

File: client-v2/src/lib/dnd/DropZone.tsx (hunk 3 above)
Comment (jonathonherbert) [reply client-v2/src/lib/dnd/DropZone.tsx]: 👍

File: client-v2/src/lib/dnd/Level.tsx (hunk 2 above)
Comment (jonathonherbert) [reply client-v2/src/lib/dnd/Level.tsx]: You're absolutely right.

**Pull request merged**